hazelcast/src/test/java/com/hazelcast/test/TestHazelcastInstanceFactory.java

```diff
@@ -281,49 +293,52 @@ public String toString() {
         return "TestHazelcastInstanceFactory{addresses=" + addressMap.values() + '}';
     }
 
-    private List<Address> createAddresses(int count) {
-        if (!isMockNetwork || count == 0) {
-            return emptyList();
-        }
-        List<Address> addresses = new ArrayList<Address>(count);
-        for (int i = 0; i < count; i++) {
-            addresses.add(createAddress());
+    private Address addNewAddressToAddressMap(int id, String host, int initialPort) {
+        synchronized (addressMap) {
+            while (true) {
+                final Address newAddress = createAddress(host, initialPort++);
+                if (!addressMap.containsValue(newAddress)) {
+                    addressMap.put(id, newAddress);
+                    return newAddress;
+                }
+            }
         }
```

Comment:
Maybe we can re-write this method like this to handle possible out of port range issue:
```
static final int MAX_PORT_NUMBER = (1 << 16) - 1;

private Address addNewAddressToAddressMap(int id, String host, int initialPort) {
    while (true) {
        int newPort = initialPort++;

        if (newPort > MAX_PORT_NUMBER) {
            throw new IllegalArgumentException("Port number is out of range: " + newPort);
        }

        Address newAddress = newAddressOrNull(host, newPort);
        if (newAddress != null) {
            synchronized (addressMap) {
                if (!addressMap.containsValue(newAddress)) {
                    addressMap.put(id, newAddress);
                    return newAddress;
                }
            }
        }
    }
}

```